Ch04/drill4.cpp

```diff
@@ -0,0 +1,76 @@
+#include "../std_lib_facilities.h"
+
+double sum(const vector<double>& v) {
+	double sumofnum = 0.0;
+	for (double i:v) {
+		sumofnum += i;
+	};
+	return sumofnum;
+}
+
+bool supportedUnit(string unit) {
+	vector<string> units = {"m", "cm", "in", "ft"};
+
+	for (string i: units) {
+		if (unit == i) {
+			return true;
+		}
+	}
+	return false;
+}
+
+double conversionFactor(string unit) {
+	if (unit == "m") {
+		return 1;
+	} else if (unit == "cm") {
+		return 1.0/100;
+	} else if (unit == "in") {
+		return 2.54/100;
+	} else {
```

Comment:
It is not clear here what the conversionFactor below is for (ft). I would be explicit with `unit == 'ft'` and remove the comparison with `m` by putting it in the `else` section.

    else {
        return 1.0;
    }